cephfs/path_test.go

```
@@ -60,3 +60,273 @@ func TestRemoveDir(t *testing.T) {
+		assert.NoError(t, err)
+
+		// Create hard link when destination exists.
+		fname2 := "TestFile2.txt"
+		f2, err := mount.Open(fname2, os.O_WRONLY|os.O_CREATE, 0666)
+		assert.NotNil(t, f2)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, f2.Close())
+			assert.NoError(t, mount.Unlink(fname2))
+		}()
+		err = mount.Link(fname1, fname2)
+		// Error, destination already exists.
+		assert.Error(t, err)
+	})
+}
+
+func TestUnlink(t *testing.T) {
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+
+	t.Run("file unlink", func(t *testing.T) {
+		fname := "TestFile.txt"
+		err := mount.Unlink(fname)
+		// Error, file does not exist.
+		assert.Error(t, err)
+
+		f, err := mount.Open(fname, os.O_WRONLY|os.O_CREATE, 0666)
+		assert.NotNil(t, f)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, f.Close())
+			assert.NoError(t, mount.Unlink(fname))
+		}()
+		assert.NoError(t, mount.Link(fname, "hardlnk"))
+
+		err = mount.Unlink("hardlnk")
+		// No Error, link will be removed.
+		assert.NoError(t, err)
+	})
+
+	t.Run("directory unlink", func(t *testing.T) {
+		dirname := "/a"
+		err := mount.MakeDir(dirname, 0755)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, mount.RemoveDir(dirname))
+		}()
+
+		err = mount.Unlink(dirname)
+		// Error, not permitted on directory.
+		assert.Error(t, err)
+	})
+}
+
+func TestSymlink(t *testing.T) {
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+
+	t.Run("file operations - source doesn't exist", func(t *testing.T) {
+		fname1 := "TestFile1.txt"
+
+		err := mount.Symlink(fname1, "Symlnk1")
+		// No Error, symlink works even if source file doesn't exist.
+		assert.NoError(t, err)
+		_, err = os.Stat(path.Join(CephMountDir, "Symlnk1"))
+		// Error, source is not there.
+		assert.Error(t, err)
+
+		localPath := path.Join(CephMountDir, fname1)
+		_, err = os.Stat(localPath)
+		// Error, source file is still not there.
+		assert.EqualError(t, err,
```

Let's avoid scraping error text if possible. os.Stat should return an error that can be tested with [os.IsNotExist](https://golang.org/pkg/os/#IsNotExist).

---

I guess this is not needed, if Stat returns an error, it means file is not there.
Depending on the error returned by Stat.